v2.2/cluster-setup-troubleshooting.md

```diff
@@ -6,202 +6,433 @@ toc: true
+
+**Solution:** Disassociate the node from the existing directory where you've stored CockroachDB data. For example, you can do either of the following:
+
+-   Choose a different directory to store the CockroachDB data:  
     {% include copy-clipboard.html %}
     ~~~ shell
-    $ pkill -9 cockroach
+    $ cockroach start --store=<new directory> --join=<cluster host> <other flags>
     ~~~
-
+-   Remove the existing directory and start a node joining the cluster again:
     {% include copy-clipboard.html %}
     ~~~ shell
-    $ rm -r testStore
+    $ rm -r cockroach-data/
     ~~~
 
-    {{site.data.alerts.callout_info}}If you're running all nodes on the same machine, skip this step. Running this command will kill your first node making it impossible to proceed.{{site.data.alerts.end}}
-
-2. On each machine, start the CockroachDB node, joining it to the first node:
-
     {% include copy-clipboard.html %}
     ~~~ shell
-    $ cockroach start --insecure --logtostderr --store=testStore \
-    --join=<host of first node>
+    $ cockroach start --join=<cluster host>:26257 <other flags>  
     ~~~
 
-    {{site.data.alerts.callout_info}}
-    If you're running all nodes on the same machine, you will need to change the <code>--port</code>, <code>--http-port</code>, and <code>--store</code> flags. For an example of this, see <a href="start-a-local-cluster.html#step-2-add-nodes-to-the-cluster">Start a Local Cluster</a>.
-    {{site.data.alerts.end}}
+### Incorrect `--join` address
+
+If you try to add another node to the cluster, but the `--join` address is not pointing at any of the existing nodes, then the process will never complete, and you'll see a continuous stream of warnings like this:
+
+~~~
+W180817 17:01:56.506968 886 vendor/google.golang.org/grpc/clientconn.go:942 Failed to dial localhost:20000: grpc: the connection is closing; please retry.  
+W180817 17:01:56.510430 914 vendor/google.golang.org/grpc/clientconn.go:1293 grpc: addrConn.createTransport failed to connect to {localhost:20000 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp [::1]:20000: connect: connection refused". Reconnecting…
+~~~
+
+**Explanation:** These warnings tell you that the node cannot establish a connection with the address specified in the `--join` flag. Without a connection to the cluster, the node cannot join.
+
+**Solution:** To successfully join the node to the cluster, start the node again, but this time include a correct `--join` address.
+
+### Missing `--join` address
 
-    Errors at this stage potentially include:
-    - The same port and host issues from [running a single node](#1-start-a-single-node-cluster).
-    - [Networking issues](#networking-troubleshooting)
-    - [Nodes not joining the cluster](#node-will-not-join-cluster)
+If you try to add another node to the cluster, but the `--join` address is missing entirely, which causes the new node to start its own separate cluster.
```

Comment:
The structure of these sentences is unclear to me. They're separate steps/conditions, so you should have some kind of structure that expresses that.

Reply:
Well, they aren't steps, they are independent conditions. Only one of them occurs at a time (AFAIK). Hence the separate sections.